storage2/src/lib.rs

```diff
@@ -1 +1,15 @@
+// Required to ensure that Rust can infer a Send bound inside the TCT
+#![recursion_limit = "256"]
 
+use jmt::WriteOverlay;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+mod metrics;
+mod state;
+mod storage;
+
+pub use crate::metrics::register_metrics;
+pub use storage::Storage;
+
+pub type State = Arc<RwLock<WriteOverlay<Storage>>>;
```

Comment:
We don't want a `State` to be an `Arc<...<`, we want it to be a new struct, whose contents are TBD based on the API we design in this PR.